tests/misc.rs

```diff
@@ -1081,6 +1081,33 @@ test! {
   stderr:   "#!/bin/sh\necho hello\n",
 }
 
+test! {
+    name: custom_tempdir,
+    justfile: r#"
+    set tempdir := "."
+    parent_tempdir := "."
+    foo:
+        #!/usr/bin/env bash
+        script_tempdir=$(basename "$(dirname $0)")
+        script_foo=`realpath {{parent_tempdir}}`/$script_tempdir/foo
+        [[ -f $script_foo ]] && echo "foo_exists" || echo "foo_missing"
+    "#,
+    stdout: "foo_exists\n",
+}
+
+test! {
+    name: standard_tempdir,
+    justfile: r#"
+    standard_parent_tempdir := "/tmp"
+    foo:
+        #!/usr/bin/env bash
+        script_tempdir=$(basename "$(dirname $0)")
+        script_foo={{standard_parent_tempdir}}/$script_tempdir/foo
+        [[ -f $script_foo ]] && echo "foo_exists" || echo "foo_missing"
+    "#,
+    stdout: "foo_exists\n",
+}
+
```

Comment:
A few comments:

- There are still a bunch of macro-based tests, but I've been moving away from them. For new tests it's best to use the `Test` struct. Check out some of the other tests.
- These tests can go in their own file, perhaps `tests/tempdir.rs`.
- `realpath` isn't available on MacOS.
- The test can be simplified with something like this:

This will only succeed if the temporary directory is the same directory as the current directory, which will only be true if the tempdir has been set:

```
set tempdir := "."
foo:
  touch $(basename "$(dirname $0)")/bar
  cat bar
```

- I think the `standard_tempdir` test can be removed.